src/test/cloudWatchLogs/commands/searchLogGroup.test.ts

```diff
@@ -5,44 +5,105 @@
 
 import * as vscode from 'vscode'
 import * as assert from 'assert'
-import { SearchLogGroupWizard, createFilterpatternPrompter } from '../../../cloudWatchLogs/commands/searchLogGroup'
+import {
+    SearchLogGroupWizard,
+    createFilterpatternPrompter,
+    TimeFilterSubmenu,
+} from '../../../cloudWatchLogs/commands/searchLogGroup'
 import { exposeEmitters, ExposeEmitters } from '../../../../src/test/shared/vscode/testUtils'
 import { InputBoxPrompter } from '../../../shared/ui/inputPrompter'
 import { createWizardTester, WizardTester } from '../../shared/wizards/wizardTestUtils'
+import { createQuickPickTester, QuickPickTester } from '../../shared/ui/testUtils'
 
 describe('searchLogGroup', async function () {
     const fakeLogGroups: string[] = []
-    let inputBox: ExposeEmitters<vscode.InputBox, 'onDidAccept' | 'onDidChangeValue' | 'onDidTriggerButton'>
-    let testPrompter: InputBoxPrompter
+    let filterPatternInputBox: ExposeEmitters<
+        vscode.InputBox,
+        'onDidAccept' | 'onDidChangeValue' | 'onDidTriggerButton'
+    >
+    let filterPatternPrompter: InputBoxPrompter
+
     let testWizard: WizardTester<SearchLogGroupWizard>
+    let testTimeRangeMenu: TimeFilterSubmenu
+
+    let customTimeRangePrompter: InputBoxPrompter
+    let defaultTimeRangePrompter: QuickPickTester<any>
+
+    let testDateBox: ExposeEmitters<vscode.InputBox, 'onDidAccept' | 'onDidChangeValue' | 'onDidTriggerButton'>
```

Comment:
We're placing a lot of variables here even though not every test uses them. You can nest `describe` blocks to avoid this:

```ts
describe('searchLogGroup', function () {
// ...
    describe('timerange submenu', function () {
        let testTimeRangeMenu: TimeFilterSubmenu
        // ...
    })
```

It's a small thing but test organization is important. Even if it does come at the cost of a little bit more boilerplate.

Oh also I'm not sure if I pointed this out but your test cases should read like `it [behavior goes here]`
Example:
```ts
    describe('Timerange Submenu', function () {
        let testTimeRangeMenu: TimeFilterSubmenu
         // test setup somewhere

         it('gives option for custom input', async function () {
             // ...
         })
    })
```
Which reads as 'Timerange Submenu gives option for custom input'

Reply:
I'll try to update some of my other PRs with this in mind. Thanks for pointing this out.